paperqa/docs.py

```diff
@@ -580,344 +580,222 @@ def delete(
-            if keys is not None:
-                texts = [t for t in texts if t.doc.dockey in keys]
-            if len(texts) == 0:
-                return
-            self.texts_index.clear()
-            self.texts_index.add_texts_and_embeddings(texts)
-        if self.jit_texts_index and keys is None:
-            # Not sure what else to do here???????
-            print(
-                "Warning: JIT text index without keys "
-                "requires rebuilding index each time!"
-            )
-            self.texts_index.clear()
-            self.texts_index.add_texts_and_embeddings(texts)
+            )[0],
+        )
+        matches = [m for m in matches if m.doc.dockey not in self.deleted_dockeys]
+        return matches[:k]
 
     def get_evidence(
         self,
-        answer: Answer,
-        k: int = 10,
-        max_sources: int = 5,
-        get_callbacks: CallbackFactory = lambda x: None,  # noqa: ARG005
-        detailed_citations: bool = False,
-        disable_vector_search: bool = False,
+        query: Answer | str,
+        exclude_text_filter: set[str] | None = None,
+        settings: MaybeSettings = None,
+        callbacks: list[Callable] | None = None,
     ) -> Answer:
         return get_loop().run_until_complete(
             self.aget_evidence(
-                answer,
-                k=k,
-                max_sources=max_sources,
-                get_callbacks=get_callbacks,
-                detailed_citations=detailed_citations,
-                disable_vector_search=disable_vector_search,
+                query=query,
+                exclude_text_filter=exclude_text_filter,
+                settings=settings,
+                callbacks=callbacks,
             )
         )
 
-    async def aget_evidence(  # noqa: C901, PLR0915
+    async def aget_evidence(
         self,
-        answer: Answer,
-        k: int = 10,  # Number of evidence pieces to retrieve
-        max_sources: int = 5,  # Number of scored contexts to use
-        get_callbacks: CallbackFactory = lambda x: None,  # noqa: ARG005
-        detailed_citations: bool = False,
-        disable_vector_search: bool = False,
+        query: Answer | str,
+        exclude_text_filter: set[str] | None = None,
+        settings: MaybeSettings = None,
+        callbacks: list[Callable] | None = None,
     ) -> Answer:
-        if len(self.docs) == 0 and self.docs_index is None:
-            # do we have no docs?
-            return answer  # type: ignore[unreachable]
-        self._build_texts_index(keys=answer.dockey_filter)
-        _k = k
-        if answer.dockey_filter is not None:
-            _k = k * 10  # heuristic - get enough so we can downselect
-        if disable_vector_search:
-            matches = self.texts
+
+        answer = Answer(question=query) if isinstance(query, str) else query
+
+        if len(self.docs) == 0 and len(self.texts_index) == 0:
+            return answer
+
+        _settings = get_settings(settings)
```

Comment:
Why do we have a private local variable name inside of a method? Imo we should just call it `settings`